index.js

```diff
@@ -1,5 +1,5 @@
 "use strict";
-const pattern = /^\s*(\d+)(?:\s*;(?:\s*url\s*=)?\s*(.+)?)?$/i;
+const pattern = /^\s*(\d+)(?:\s*;(?:\s*url\s*=)?\s*(?:("|')\s*(.+?)?\s*\2|(.+?))?)?\s*$/i;
```

Comment:
Wouldn't changing the `(.+?)` to just `.` be the same here?

Reply:
I do not follow. There are two instances of `(.+?)` and both are needed to _lazily_ capture one or more URL chars. The first captures chars inside quotes (single or double) and the second captures non-quoted chars. Regex's are not fun so you may see something that I don't.

Reply:
Lazily capturing "one or more" sounds the same as capturing "zero or more" (`.*`).

Reply:
Ah, I now see your point. The `*` didn't come through in your initial comment. I have cleaned up the regex with e7543c22ff6914302b75b76032e6d55e900a2e12. We still need the lazy quantifiers though.

Reply:
What do we need the lazy quantifiers for, exactly?

Reply:
If we drop the `?` then it will greedily take trailing white spaces. You can verify this by using the modified pattern `^\s*(\d+)(?:\s*;(?:\s*url\s*=)?\s*(?:("|')\s*(.*)\s*\2|(.*)))?\s*$` and see that it fails this test `"5; url= http://domain.com/ "`.